app/assets/javascripts/components/MentionInput.jsx

```diff
@@ -83,30 +83,61 @@ class MentionInput extends React.Component {
     }
 
     handleKeyPress = (event) => {
-      cursor = this.textInput.selectionStart;
-      const curr = this.textInput.value + event.key;
-      const closestAt = curr.lastIndexOf('@', cursor);
+      if (this.props.mention) {
+        // Comment with mention
+        cursor = this.textInput.selectionStart;
+        const curr = this.textInput.value + event.key;
+        const closestAt = curr.lastIndexOf('@', cursor);
+
+        // if user types in '@'
+        if (event.charCode == 64) {
+          this.setState({
+            showUsers: true,
+            users: Array.from(this.state.originalUsers)
+          });
+        }
 
-      // if user types in '@'
-      if (event.charCode == 64) {
+        if (this.state.showUsers) {
+          if (closestAt != -1) {
+            const currentTypedUser = curr.substring(closestAt + 1, cursor + 1);
+            this.filterUsersByName(currentTypedUser)
+          }
+        }
+      } else {
+        // Search for user, no "@" required
+        const curr = this.textInput.value + event.key;
```

Comment:
small thing: if `curr` has same value in both if and else blocks, could this be pulled outside of both blocks?